src/vmm/src/persist.rs

```diff
@@ -548,22 +584,44 @@ fn guest_memory_from_uffd(
         .create()
         .map_err(GuestMemoryFromUffdError::Create)?;
 
-    for mem_region in guest_memory.iter() {
-        uffd.register(mem_region.as_ptr().cast(), mem_region.size() as _)
-            .map_err(GuestMemoryFromUffdError::Register)?;
-    }
+    let fds = match guest_memfd_fd {
+        Some(guest_memfd) => {
+            guest_memory_uffd_register_minor(&guest_memory, &uffd);
+            vec![
+                uffd.as_raw_fd(),
+                guest_memfd,
+                userfault_bitmap_memfd
+                    .expect("memfd is not present")
+                    .as_raw_fd(),
+            ]
+        }
+        None => {
+            for mem_region in guest_memory.iter() {
+                uffd.register(mem_region.as_ptr().cast(), mem_region.size() as _)
+                    .map_err(GuestMemoryFromUffdError::Register)?;
+            }
+            vec![uffd.as_raw_fd()]
+        }
+    };
```

Comment:
nit: something like the following is a bit shorter and less repetetive

```rs
let mut mode = RegisterMode::Missing;
let mut fds = vec![uffd.as_raw_fd()];

if let Some(gmem) = guest_memfd_fd {
    mode = RegisterMode::from_bits_retain(UFFDIO_REGISTER_MODE_MINOR);
    fds.push(gmem);
    fds.push(userfault_bitmap_memfd);
}

for mem_region in guest_memory.iter() {
    uffd.register_with_mode(mem_region.as_ptr().cast(), mem_region.size() as _, mode)
        .map_err(GuestMemoryFromUffdError::Register)?;
}
``

Reply:
applied, thanks